### ISSUE TITLE
change of the URL, when you click on a seperate file in the menu

### ISSUE THREAD

**Issue by secure-77** [secure-77/Perlite#11]
I need to implement and test this

Addition in perlite.js
```js
function changeURL(str){
  window.history.pushState({},"", "/?link="+str);
  getContent(str);
}
```
Change in helper.php -> menu function
```php
$html .= '		<li>
                                    <a href="#" onclick=changeURL("'. $pathClean .'"); id="'. $pathID .'" class="perlite-link">'. $mdFile .'</a>
                                </li>';
```

_Originally posted by @gigibu5 in https://github.com/secure-77/Perlite/issues/10#issuecomment-1072964640_

**Comment by gigibu5** [comment]
If this is implemented, I also recommend implementing this return of the translateLink function in content.php:
```php
//return '<a href="?link='.$urlPath.'">'. $linkName .'</a>'; // old return
return '<a href="#" onclick=changeURL("'.$urlPath.'");>'.$linkName.'</a>'; // new return
```
With this, the page won't reload on links from within files.

**Comment by secure-77** [comment]
Implemented in Version 1.4.2
